### DISCUSSION TITLE
Expanding target mod

### DISCUSSION THREAD

**Discussion by rasputozen** [ppy/osu#15125] [Ideas] [answered]
Target mod is essentially an auto-map generator for a song. Would be cool to see this expanded to be able to generate more complex and varied auto-maps. Sometimes there are no good maps for a song you like and this would be a way to quickly generate a high quality one as a starting point for a new map or as an adequate replacement altogether.

**Comment by bdach** [comment, marked as answer]
Generating a (good quality) map for an arbitrary song could be a separate multi year research project, and we're not even anywhere near being done with the actual game part of the game. Sorry, but this proposal is too general and far reaching in scope to be realistic or useful.

**Reply by peppy** [reply]
Random mod is probably the closest we'll ever go to auto generated maps, and closer to what you seen to be after.

**Comment by rasputozen** [comment]
I can see how difficult and involved it would be now that you mention it. Would a small scripting language where users could write their own target-generating algorithms and put them in a folder that osu scans on startup be more reasonable?

**Reply by bdach** [reply]
no. that is also out of scope. if you wish to generate maps you can use the lazer codebase itself and the beatmap encoder shipped within to generate your maps.

**Reply by rasputozen** [reply]
Fair enough.
